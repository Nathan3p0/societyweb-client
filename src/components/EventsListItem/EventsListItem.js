import React from 'react';
import { Link } from 'react-router-dom';

const EventsListItem = (props) => {
    return (
        <Link to={`/admin/events/${props.id}`}>
            <li className="tableRow">
                <div className="col col-1" data-label="Event Date">{props.date}</div>
                <div className="col col-2" data-label="Event Name">{props.name}</div>
                <div className="col col-3" data-label="Event Time">{props.time}</div>
                <div className="col col-4" data-label="Customer Phone">{props.location}</div>
            </li>
        </Link>
    );
}

EventsListItem.defaultProps = {
    id: 0,
    date: '12/12/79',
    name: 'Default Name',
    time: '11:11:11',
    location: 'Default Location'
}

export default EventsListItem;