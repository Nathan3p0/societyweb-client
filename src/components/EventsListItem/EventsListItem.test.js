import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import EventsListItem from './EventsListItem'
import { BrowserRouter } from 'react-router-dom'

describe('Events List item Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <BrowserRouter>
                <EventsListItem />
            </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <EventsListItem />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    })
})