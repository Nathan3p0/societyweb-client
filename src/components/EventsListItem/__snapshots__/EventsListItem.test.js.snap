// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events List item Component Rendering Tests renders to UI as expected 1`] = `
<a
  href="/admin/events/0"
  onClick={[Function]}
>
  <li
    className="tableRow"
  >
    <div
      className="col col-1"
      data-label="Event Date"
    >
      December 12th 1979
    </div>
    <div
      className="col col-2"
      data-label="Event Name"
    >
      Default Name
    </div>
    <div
      className="col col-3"
      data-label="Event Time"
    >
      11:11:11
    </div>
    <div
      className="col col-4"
      data-label="Customer Phone"
    >
      Default Location
    </div>
  </li>
</a>
`;
